chill-protobuf/src/main/java/com/twitter/chill/protobuf/ProtobufSerializer.java

```diff
@@ -43,21 +43,32 @@
    * classes in play, which should not be very large.
    * We can replace with a LRU if we start to see any issues.
    */
+  // Cache for the `parseFrom(byte[] bytes)` method
   final protected HashMap<Class, Method> methodCache = new HashMap<Class, Method>();
+  // Cache for the `getDefaultInstance()` method
+  final protected HashMap<Class, Method> defaultInstanceMethodCache = new HashMap<Class, Method>();
 
   /**
    * This is slow, so we should cache to avoid killing perf:
    * See: http://www.jguru.com/faq/view.jsp?EID=246569
    */
-  protected Method getParse(Class cls) throws Exception {
-    Method meth = methodCache.get(cls);
+  private Method getMethodFromCache(Class cls, HashMap<Class, Method> cache, String methodName, Class... parameterTypes) throws Exception {
+    Method meth = cache.get(cls);
     if (null == meth) {
-      meth = cls.getMethod("parseFrom", new Class[]{ byte[].class });
-      methodCache.put(cls, meth);
+      meth = cls.getMethod(methodName, parameterTypes);
+      cache.put(cls, meth);
     }
     return meth;
   }
 
+  protected Method getParse(Class cls) throws Exception {
+    return getMethodFromCache(cls, methodCache, "parseFrom", new Class[]{byte[].class});
```

Comment:
one thing I just noticed: we unconditionally allocate this every call now, in the past we only hit if once (when the cache was empty).

Can we make a private variable to hold this and allocate it only one time?

Reply:
hmm, I could also do:
```java
return getMethodFromCache(cls, methodCache, "parseFrom", byte[].class);
```

Reply:
lets do that.

Reply:
Done